tests/app/main/test_frameworks.py

```diff
@@ -4284,6 +4277,30 @@ def test_upload_signature_page(
         assert res.status_code == 302
         assert res.location == 'http://localhost/suppliers/frameworks/g-cloud-8/234/contract-review'
 
+    @mock.patch('dmutils.s3.S3')
+    @mock.patch('app.main.views.frameworks.file_is_empty')
+    def test_signature_upload_returns_400_if_no_file_is_chosen(
+        self, file_is_empty, s3, return_supplier_framework, data_api_client
+    ):
+        self.login()
+
+        data_api_client.get_framework.return_value = get_g_cloud_8()
+        data_api_client.get_framework_agreement.return_value = self.framework_agreement()
+        return_supplier_framework.return_value = self.supplier_framework(
+            framework_slug='g-cloud-8',
+            on_framework=True
+        )['frameworkInterest']
+        s3.return_value.get_key.return_value = None
```

Comment:
I assume this set up is just copied from another test but a small thing that might improve readability would be an inline comment saying why we are setting the s3 return value as such ie. `no existing file`

Reply:
Yep this follows the same pattern from other tests on this route.